vsphere/datadog_checks/vsphere/vsphere.py

```diff
@@ -474,6 +474,158 @@ def query_metrics_wrapper(self, query_specs):
+            self.gauge(
+                'vsphere.vsan.cluster.time',
+                t0.total(),
+                tags=self._config.base_tags,
+                raw=True,
+                hostname=self._hostname,
+            )
+
+            for cluster in new_health_metrics:
+                for metric_name, metric_value in cluster.items():
+                    list_of_tags = ["{}:{}".format(key, value) for key, value in metric_value.items()]
+                    self.gauge(
+                        metric_name,
+                        1,
+                        tags=list_of_tags + self._config.base_tags,
+                        raw=True,
+                        hostname=self._hostname,
+                    )
+
+            for cluster in new_performance_metrics:
+                for entity_type in cluster:
+                    if len(entity_type.value) > 0:
+                        resource_type = entity_type.value[0].metricId.dynamicProperty[0][0]
+                        for given_metric in entity_type.value:
+                            self.log.debug(
+                                "Processing metric with type:%s",
+                                type(given_metric),
+                            )
+                            latest_value = given_metric.values.split(',')[-1]
+                            if latest_value != 'None':
+                                if given_metric.metricId.label in VSAN_PERCENT_METRICS:
+                                    latest_value = float(latest_value) / 100.0
+                                if resource_type == 'cluster':
+                                    if 'vsan.cluster.{}'.format(given_metric.metricId.label) in VSAN_METRICS:
+                                        self.gauge(
+                                            'vsphere.vsan.cluster.{}'.format(given_metric.metricId.label),
+                                            # for now we only collect the latest value
+                                            float(latest_value),
+                                            tags=[
+                                                'vsphere_cluster:{}'.format(given_metric.metricId.dynamicProperty[0][1])
+                                            ]
+                                            + self._config.base_tags,
+                                            raw=True,
+                                            hostname=self._hostname,
+                                        )
+                                    else:
+                                        self.log.debug(
+                                            "Skipping metric %s because it is not in the list of metrics to collect",
+                                            given_metric.metricId.label,
+                                        )
+                                elif resource_type == 'host':
+                                    if 'vsan.host.{}'.format(given_metric.metricId.label) in VSAN_METRICS:
```

Comment:
If we make `'vsan.host.{}'.format(given_metric.metricId.label)` a constant, we can use this as the metric name if we remove `raw=True` in the `self.gauge` method

Reply:
Interesting, I didn't know that you could do this! I've made the change.